### PR TITLE
[tests-only][full-ci] added test to reshare a share invitation to other user

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
In this PR, added test to reshare a share invitation to other user.

Added Scenarios:
```feature
Scenario Outline: reshare a file to user with different roles
Scenario Outline: reshare a folder to user with different roles
Scenario Outline: reshare a file inside project space to a user with different roles
Scenario Outline: reshare a folder inside project space to a user with different roles
```
Currently, user with a role with no sharing permission trying to reshare the share invitation returns `500` status code, mentioned in this issue https://github.com/owncloud/ocis/issues/8112. Those tests will be added in next PR. 

## Related Issue

- https://github.com/owncloud/ocis/issues/7947

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/reshare.feature [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+      | Carol    |
+
+
+  Scenario Outline: reshare a share invitation for file to user with different roles
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/reshare.feature:16]:
```suggestion
  Scenario Outline: reshare a file to a user with different roles
```

File: tests/acceptance/features/apiSharingNg/reshare.feature [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      | Co Owner    | Viewer       |
+      | Co Owner    | File Editor  |
+      | Co Owner    | Co Owner     |
+      | Manager     | Viewer       |
+      | Manager     | File Editor  |
+      | Manager     | Co Owner     |
+      | Manager     | Manager      |
+
+
+  Scenario Outline: reshare a share invitation for folder to user with different roles
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNg/reshare.feature:109]:
```suggestion
  Scenario Outline: reshare a folder to a user with different roles
```

**Review by saw-jan — APPROVED** [review]

File: tests/acceptance/features/apiSharingNg/reshare.feature [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+      | role         | <role>        |
+    When user "Brian" sends the following share invitation using the Graph API:
+      | resourceType | file           |
+      | resource     | textfile1.txt  |
+      | space        | Shares         |
+      | sharee       | Carol          |
+      | shareType    | user           |
+      | role         | <reshare-role> |
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should match
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/reshare.feature:33]: If there're any differences in resharing then we must do the response check or check only the changes. If not then maybe we can only check the reshared resource by resharee

[the remark this answers is not in the document]
Comment (Salipa-Gurung) [reply tests/acceptance/features/apiSharingNg/reshare.feature:33]: @PrajwolAmatya could you follow up this discussion?

**Review by SagarGi — APPROVED** [review]
LGTM as well :+1:

**Pull request merged**